packages/backend/server/src/core/doc/event.ts

```diff
@@ -1,17 +1,35 @@
 import { Injectable } from '@nestjs/common';
 
 import { OnEvent } from '../../base';
+import { Models } from '../../models';
 import { DocReader } from './reader';
 
 @Injectable()
 export class DocEventsListener {
-  constructor(private readonly doc: DocReader) {}
+  constructor(
+    private readonly docReader: DocReader,
+    private readonly models: Models
+  ) {}
 
   @OnEvent('doc.snapshot.updated')
   async markDocContentCacheStale({
     workspaceId,
     docId,
   }: Events['doc.snapshot.updated']) {
-    await this.doc.markDocContentCacheStale(workspaceId, docId);
+    await this.docReader.markDocContentCacheStale(workspaceId, docId);
+    const snapshot = await this.models.doc.get(workspaceId, docId);
+    if (!snapshot) {
+      return;
+    }
+
+    const isDoc = workspaceId !== docId;
+    // update doc content to database
+    if (isDoc) {
+      const content = this.docReader.parseDocContent(snapshot.blob);
```

Comment:
do it in a standalone job? this event is emitted right after doc gets merged. doing such heavy job will hold cpu too long

Reply:
or we might send the snapshot binary in event payload, to avoid read it from db again

Reply:
Okay, I'll make some changes